### PR TITLE
component tests / snapshots

### PR DESCRIPTION
**Pull request by luetkemj**
- Add enzyme for component tests
- move jest to dev dependency
- import props from ‘prop-types’ so react will stop yelling at me
- component snapshots

### CODE REVIEW

**Comment by luetkemj** [comment]
Had some interesting issues with the build on this one. React is required as a dev dependency and the snapshots got out of date on travis but not on my machine - which is odd. You should probably test this locally before merging.

File: test/src/HexGrid.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import HexGrid from '../../src/HexGrid';
+
+test('HexGrid should render correctly with default props', () => {
+  const tree = renderer.create(
+    <HexGrid>
+      <div>child</div>
```
Comment (Hellenic) [inline test/src/HexGrid.test.js]: <div> is not allowed in SVG structure but for testing purposes I guess it doesn't matter?

**Comment by Hellenic** [comment]
Lots of tests, thank for this! Yeah, I'll run it locally too before merging

**Comment by luetkemj** [comment]
I think having it in peer and dev is ok. In production it shouldn't be included in the package. I've heard conflicting opinions on peer deps but I honestly don't have an opinion yet.

File: test/src/HexGrid.test.js (hunk 1 above)
Comment (luetkemj) [inline test/src/HexGrid.test.js]: Yeah, that was my thought. Props just wanted a node.

**Comment by Hellenic** [comment]
Yeah, I was just reading about that too, and I think what's there currently is good actually. We should just check the Travis config to have it install all the deps correctly

**Comment by Hellenic** [comment]
I just now ran your fork's test on my local, and get the same as TravisCI. 
Snapshot Summary
 › 2 snapshots written in 1 test suite.
 › 1 obsolete snapshot file found, run with `npm test -- -u` to remove it.
And it fails.

I can also try to look more into it but won't be before the weekend

**Comment by luetkemj** [comment]
@Hellenic Any update?

**Comment by Hellenic** [comment]
Hey!
No sorry, wasn't able to dig into it yet. Have you tried if you take a clean copy of your own fork and run the tests, does it still work?

**Comment by Hellenic** [comment]
And to be more precise: So the build & tests are passing, so that part is fine. It's still telling stuff about the snapshots though. And that's the same I got on my local.

So anyway if you can try, if you get the same. At least from what I understand about snapshots, they should not be written each time, otherwise they won't test anything

**Comment by luetkemj** [comment]
Just installed a clean clone from master of my fork. Ran tests and got a clean bill of health.

>At least from what I understand about snapshots, they should not be written each time, otherwise they won't test anything

that is my understanding as well. I think what we are experiencing is related to this "issue": https://github.com/wallabyjs/public/issues/1074

Running the command `npm test -- -u` seems to clean out the obsolete snapshots. Try running that and see what you get. That said, I don't think the command should be a normal part of the travis script but was added as a temp fix only and can be removed.

screen shot 2017-05-18 at 9 01 59 am

File: src/HexGrid.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import React, { Component, PropTypes } from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
```
Comment (Hellenic) [inline src/HexGrid.js]: Should `prop-types` be included in the (dev/peer)dependencies?

Comment (luetkemj) [inline src/HexGrid.js]: good catch - added to peer deps alongside react.

**Comment by Hellenic** [comment]
I'm gonna go ahead and merge, and if needed I'll tune the test running so that it works correctly with Travis

**Pull request merged**